onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_support_checker.cc

```diff
@@ -0,0 +1,1175 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <core/common/logging/logging.h>
+#include <core/common/safeint.h>
+#include <core/providers/common.h>
+
+#include <core/graph/graph.h>
+
+#include "helper.h"
+#include "op_support_checker.h"
+
+namespace onnxruntime {
+namespace nnapi {
+
+using std::vector;
+
+#pragma region helpers
+
+bool HasExternalInitializer(const InitializerMap& initializers, const Node& node) {
+  for (const auto* node_arg : node.InputDefs()) {
+    const auto& input_name(node_arg->Name());
+    if (!Contains(initializers, input_name))
+      continue;
+
+    const auto& tensor = initializers.at(input_name);
+    if (tensor.has_data_location() &&
+        tensor.data_location() == ONNX_NAMESPACE::TensorProto_DataLocation_EXTERNAL) {
+      LOGS_DEFAULT(VERBOSE) << "Initializer [" << input_name
+                            << "] with external data location are not currently supported";
+      return true;
+    }
+  }
+
+  return false;
+}
+
+#pragma endregion helpers
+
+#pragma region op_base
+
+class BaseOPSupportChecker : public IOPSupportChecker {
+ public:
+  virtual ~BaseOPSupportChecker() = default;
+  bool IsOpSupported(const InitializerMap& initializers, const Node& node,
+                     const OPSupportCheckParams& params) override;
+
+ protected:
+  virtual bool IsOpSupportedImpl(const InitializerMap& /* initializers */, const Node& /* node */,
+                                 const OPSupportCheckParams& /* params */) {
+    return true;
+  }
+  virtual int32_t GetMinSupportedSdkVer(const Node& /* node */, const OPSupportCheckParams& /* params */) const {
+    return 27;
+  }
+
+  virtual bool HasSupportedInputs(const Node& node);
+
+  virtual int GetMinSupportedOpSet(const Node& /* node */) { return 1; }
+  virtual int GetMaxSupportedOpSet(const Node& /* node */) { return 13; }
+  bool HasSupportedOpSet(const Node& node);
+};
+
+bool BaseOPSupportChecker::IsOpSupported(const InitializerMap& initializers, const Node& node,
```

Comment:
>BaseOPSupportChecker [](start = 5, length = 20)

nit: 'OP' being all in uppercase implies it's an abbreviation. if it's not, *OpSupportChecker would be mildly better.